repair_shards.c

```diff
@@ -98,9 +102,12 @@ master_copy_shard_placement(PG_FUNCTION_ARGS)
 	}
 
 	HOLD_INTERRUPTS();
-
-	/* TODO: implement row copying functionality */
-	ereport(ERROR, (errmsg("shard placement repair not fully implemented")));
+	dataCopied = CopyDataFromFinalizedPlacement(sourcePlacement, targetPlacement,
```

Comment:
1. Yup.
2. I'll look over all error messages shortly and make them fit with conventions.